### ISSUE TITLE
Stop using `<root_prefix>/profile.d/`

### ISSUE THREAD

**Issue by jonashaag** [mamba-org/mamba#2524]
IMO we should get rid of the feature that writes shell activation scripts to `<root_prefix>/profile.d/`. Instead we should just source from the output of the Mamba command directly. I don't think many people are using the `profile.d/` feature to customize their setup.

**Comment by jjerphan** [comment]
I also agree that such a script (inherited from the original design of `conda`) is not necessary to be able to use `mamba`.

To me, this adds more complexity than what it removes, and I am in favor of removing the script entirely with a deprecation period lasting until a later version of `mamba`, for instance 2.2.

I think the warning needs to be adapted in this regard to point to this issue for discussions.

**Comment by qiyubio** [comment]
Does that mean `source <root_prefix>/profile.d/conda.sh` would be enough for mamba/2.0.7 and forward, also compatible with lower version of mamba too? 
As HPC staff, we want to make sure minimal impact on users' side given the large diversity of mamba version they had.

**Comment by minrk** [comment]
I don't necessary object to the removal of the script, but the warnings seem to be a problem, both not actionable and inaccurate. The check for `${MAMBA_ROOT_PREFIX}` seems to get resolved during installation to `if [-z ""]` so the condition is always true (at least for conda-forge mamba package 2.1.1), showing the warning regardless of whether the variable is set.

FWIW, I do think the user experience of the profile.d script is nicer than the more complex invocation, consistent with many other shell tools (bash_completion, etc.) and it means mamba can update its setup to work over time if the args should change, unlike the more complex, less obvious direct invocation of `mamba shell hook`.

**Comment by jjerphan** [comment]
I having this script in the first place was an error to me, and it comes with a lot of complexity for what it brings in practice.

Honestly, I am really annoyed by it, but probably the cost of its removal is greater than the cost of keeping it. I would be in favor of removing the warning and moving on with it.

**Comment by ndevenish** [comment]
Also `mamba shell init` is explicitly the wrong thing to tell people to do if they are e.g. trying to use mamba inside a shell script launched in CI or on an HPC system, which at least in my experience (or our and other HPC sites) is the majority of instances of people here actually sourcing this script directly. Running this command won't fix the warning or the problem, and might end up breaking their login environments, as IIRC `auto_activate_base: true` is the default[^1]? (second most common usage: People with old profile scripts from when this stuff was a lot less stable). 

I guess the actual replacement suggestion for this case should be `$(/full/path/to/shared/mamba shell hook)`? Or is `etc/profile.d/conda.sh` still going to be present and should be used? What is the actual long-term-promised-api way one is supposed to do this?

This change and the current warning seems to have been put in without a lot of discussion[^2]. At the very least I'd suggest backing it out and working out a smoother way to deprecate this; Hyrum's law may have made this too entwined for a quick and surgical removal.

[^1]: Most of our users don't load mamba directly themselves but via `module load mamba` on our [environment modules](https://modules.readthedocs.io/en/latest/) system, so anyone not loading-at-login would not have the default-activation behaviour affecting their environment or PS1. We rolled back the mamba update, but any user logging in who had added this as a load-on-login was getting this big warning.

[^2]: The title of this discussion ticket seems even wrong; At least in my installations it's in `<root_prefix>/etc/profile.d/`, although I can't eliminate the possibility that my environments are weird/out of date and that's where it is now put.

**Comment by jjerphan** [comment]
This script will eventually not be removed, see #3946.
